base_layer/core/src/proof_of_work/diff_adj_manager/diff_adj_storage.rs

```diff
@@ -186,6 +206,8 @@ where T: BlockchainBackend
         if let Some((sync_height, _)) = self.sync_data {
             for height in (sync_height + 1)..=height_of_longest_chain {
                 let header = self.blockchain_db.fetch_header(height)?;
+                // add new timestamps
+                self.timestamps.push_back(header.timestamp);
```

Comment:
Instead of the removing the elements from the front using the while loop at ln227 to ln229, maybe use an if statement here:
`if self.timestamps.len() > 11 {
                self.timestamps.remove(0);
  }` 
It shouldn't be able to grow past the max length.

Reply:
Done